node/src/neighborhood/mod.rs

```diff
@@ -3644,45 +3651,138 @@ mod tests {
-        replacement_database
-            .add_arbitrary_full_neighbor(subject_node.public_key(), relay1.public_key());
-        replacement_database.add_arbitrary_full_neighbor(relay1.public_key(), relay2.public_key());
-        replacement_database.add_arbitrary_full_neighbor(relay2.public_key(), exit.public_key());
-        subject.gossip_acceptor_opt = Some(Box::new(DatabaseReplacementGossipAcceptor {
+        let mut nodes = vec![
+            subject_node.public_key().clone(),
+            relay1.public_key().clone(),
+        ];
+        let mut neighborhood: Neighborhood = neighborhood_from_nodes(&subject_node, Some(&relay1));
+        let mut replacement_database = neighborhood.neighborhood_database.clone();
+
+        fn nonce(x: u16) -> u16 {
+            x + (10 * x) + (100 * x) + (1000 * x)
+        }
+
+        for i in 1..=hops {
+            match i {
+                1 => {
+                    replacement_database.add_node(relay1.clone()).unwrap();
+                }
+                i if i < hops => {
+                    let relay_node = make_node_record(nonce(i), false);
+                    replacement_database.add_node(relay_node.clone()).unwrap();
+                    nodes.push(relay_node.public_key().clone());
+                }
+                i if i == hops => {
+                    let exit_node = make_node_record(nonce(i), false);
+                    replacement_database.add_node(exit_node.clone()).unwrap();
+                    nodes.push(exit_node.public_key().clone());
+                }
+                _ => panic!("The match statement should be exhausting."),
+            }
+            replacement_database
+                .add_arbitrary_full_neighbor(&nodes[i as usize - 1], &nodes[i as usize]);
+        }
+
+        neighborhood.gossip_acceptor_opt = Some(Box::new(DatabaseReplacementGossipAcceptor {
             replacement_database,
         }));
-        subject.persistent_config_opt = Some(Box::new(
+        neighborhood.persistent_config_opt = Some(Box::new(
             PersistentConfigurationMock::new().set_past_neighbors_result(Ok(())),
         ));
-        subject
-            .overall_connection_status
-            .update_can_make_routes(false);
+
+        neighborhood
+    }
+
+    #[test]
+    fn neighborhood_starts_accountant_when_first_route_can_be_made() {
         let (accountant, _, accountant_recording_arc) = make_recorder();
-        let system = System::new("neighborhood_does_not_start_accountant_if_no_route_can_be_made");
+        let (ui_gateway, _, _) = make_recorder();
+        let mut subject = make_neighborhood_linearly_connected_with_nodes(3);
+        subject.node_to_ui_recipient_opt = Some(ui_gateway.start().recipient());
         let peer_actors = peer_actors_builder().accountant(accountant).build();
         bind_subject(&mut subject, peer_actors);
+        let system = System::new("neighborhood_does_not_start_accountant_if_no_route_can_be_made");
```

Comment:
System name should either be something simple like "test" or the same as the test name.